cpp/src/parquet/statistics.cc

```diff
@@ -552,12 +560,19 @@ class TypedStatisticsImpl : public TypedStatistics<DType> {
 
   void Merge(const TypedStatistics<DType>& other) override {
     this->num_values_ += other.num_values();
-    if (other.HasNullCount()) {
+    // Merge always runs when Merge builder's page statistics
+    // into column chunk statistics, so it tent to have null count.
+    if (ARROW_PREDICT_TRUE(other.HasNullCount())) {
       this->statistics_.null_count += other.null_count();
+    } else {
+      this->has_null_count_ = false;
     }
-    if (other.HasDistinctCount()) {
-      this->statistics_.distinct_count += other.distinct_count();
+    // Distinct count cannot be merged.
```

Comment:
I understand the reasoning, but this might break existing usage. Perhaps we should instead document that the distinct count can be an upper bound?

Can you use `git log` or `git blame` to find out who touched this code previously, and ping them to get their opinion?

Reply:
https://github.com/apache/arrow/commit/43c7154894f819e617dbf3d669dbe1185f05893b#diff-f8a090a40e214aba5ebff534411a9febf213b725597afb750395d2a7483d9bc1

I guess these code is written in 2016 and never changed the syntax for add distinct count. Maybe it's designed like dictionary in Arrow IPC Streaming?

Reply:
Hmm... ok. I took a look at parquet-mr and the Rust Parquet implementation, neither of them seems to handle `distinct_count`. So I guess this is ok.

Reply:
I can discard it and add support for it using `set` later.